### DISCUSSION TITLE
How Do I Have A Custom Snippet In Helix.

### DISCUSSION THREAD

**Discussion by mubin-thinks** [helix-editor/helix#5327] [Q&A]
C++ is my main programming language. And I need to add a custom snippet. How do I add it?
Hoping to get my answer soon.

**Comment by the-mikedavis** [comment]
Snippets are not currently supported. https://github.com/helix-editor/helix/issues/395 covers adding them. There's a `snippets` branch working on LSP snippets but it isn't complete yet. Once LSP snippets are implemented, custom snippets could be added re-using some of that new code.

**Reply by astro-ray** [reply]
Has development on snippets stopped? I couldn't find the `snippet` branch.

**Reply by the-mikedavis** [reply]
The changes in that branch were merged into master in https://github.com/helix-editor/helix/pull/5864

**Comment by FergusFettes** [comment]
Afaik snippets have been implemented by now, but I'm still not 100% sure how to use them. I saw someone do it somewhere in the github issues but can't find it again.

I have this:
```
[language.config.completion.snippets.custom.ipdb]
prefix = ["ipdb"]
body = ["__import__('ipdb').set_trace()"]
description = "Insert IpDB call."
scope = "expr"
```
in my languages.toml but I don't see anything.

**Reply by pascalkuthe** [reply]
we only support the snippts send by the LS with no way to configure you own beyond that. Some LS support custom snippets in their own config but not all do and the config options are not uniform. Refer to the documentation of the specific LS for the specific config option

**Comment by quantonganh** [comment]
https://github.com/helix-editor/helix/issues/395#issuecomment-1671394603
